packages/components/src/common/domHelper.js

```diff
@@ -1,33 +1,89 @@
-export function getPostByElement(posts, elementToFind) {
-  return posts.find(post => post.$el === elementToFind);
+export function getFeedElement(insaneMode = false) {
+  const feedClass = insaneMode ? "feed__insane" : "feed__cards";
+
+  return document.getElementsByClassName(feedClass)[0];
 }
 
-export function getBelowPost(el) {
-  if (el.nextElementSibling === null) return el;
+export function getCardsPerRow() {
+  const feed = getFeedElement();
 
-  return el.nextElementSibling;
+  if (!feed) return 0;
+
+  let previousTop = 0;
+  let cardsPerRow = 0;
+
+  for (let i = 0; i < feed.childNodes.length; i += 1) {
+    const currentTop = feed.childNodes[i].offsetTop;
+
+    if (previousTop !== currentTop && previousTop > 0) return cardsPerRow + 1;
+
+    if (previousTop === currentTop) cardsPerRow += 1;
+
+    previousTop = currentTop;
+  }
+
+  return 0;
 }
 
-export function getAbovePost(el) {
-  if (el.previousElementSibling === null) return el;
+export function getElementIndexFromSiblings(targetElement) {
```

Comment:
Can we try to think about an implementation that doesn't require looping through all previous elements to calculate the index? I fear that in extreme cases this loop can slow things down.
Maybe we can save the index of the selected post instead of saving the reference to the element. I'm open to any other ideas. Let me know.

Reply:
Yes I think we can do that. I'll probably create a map when they are rendered so we won't have to recompute everytime, identifying of the index will be part of the render itself that will have an O(1) when finding and saving its index.

Reply:
Yes, I'm looking for an O(1) solution. That will be perfect. I'm here if you want to discuss it over.

Reply:
Will definitely ping you up once I hit a brick wall 😄 Thank you Ido!

Reply:
Hi @idoshamun, what do you think if I expand the `hoveredPost` reference in making it an array having the current implementation value at index `0` and the actual index of the element at index `1` so it won't have to recompute every time?

Reply:
That sounds interesting! We can try it out

Reply:
Will create a PR soon!